### ISSUE TITLE
 Pay Transaction Fees with Different Inputs

### ISSUE THREAD

**Issue by raspi0124** [CounterpartyXCP/counterparty-lib#1039]
I saw that there are similar issues already, but its solution doesn't fit me, so I'm opening a new issue.
How can I do something like creating a transaction with from-address of  "a" to receiver address "c" but address "b" pays for a fee?
I'm stuck here.

So, I want to do something like this.
"A" sends token to "C", and "A" will not pay for transaction fee but the other address "B" will pay the transaction fee.
I own both "A" address and "B" address.

**Comment by droplister** [comment]
I don't believe what you've explained is a feature of Bitcoin. There are things like replace by fee (RBF) and child-pays-for-parent (CPFP), neither of which any Counterparty wallets support, to my knowledge. But 
 even if they did, address "A" would have to be the one generating such txs.

What are you trying to accomplish/avoid in your design?

**Comment by raspi0124** [comment]
Well, I'm using an account system on coind, and since coind's account system doesn't keep balance in each address, but put multiple account's balance on one address, when I try to send token from the account's address (this address doesn't have any BTC in it since it automatically moved to other address by coind's account system), the api server returned me following
"Error composing send transaction via API: Insufficient MONA at address XXXXXXXXXXXXXXX. (Need approximately 0.001546 MONA.) To spend unconfirmed coins, use the flag `--unconfirmed`. (Unconfirmedcoins cannot be spent from multi\u2010sig addresses.)"

**Comment by droplister** [comment]
Hmm, I would address this issue to the Monaparty developers because balances moving automatically between addresses is not a behavior exhibited by Counterparty.

**Comment by deweller** [comment]
The counterparty protocol supports multiple inputs.  The order matters, as the first input must be the address sending the token.  So to accomplish what you want, do the  following:

Input 0: Address "A" pays 0.0001 BTC
Input 1: Address "B" pays 0.0001 BTC

Output 0: 0 BTC OP_RETURN which encodes the send to address "C"
Output 1: 0.0001 BTC paid to address "A"

In this case, address "A" gets the 0.0001 BTC returned.  And address "B" paid the fee of 0.0001 BTC.

It is required, though, that address "A" provide some amount of BTC to the transaction and that input must be the first input.  This is how the protocol determines what address the tokens are sent from.

**Comment by chiguireitor** [comment]
Excellent explanation Devon.

What this means is that if you can control precisely the TX construction, you can do it.... however, if you're not that technical, you just can solve your issue sending some funds to the address.

**Comment by cryptcoin-junkey** [comment]
Monacoin is almost same as Bitcoin. So the comment by @deweller https://github.com/CounterpartyXCP/counterparty-lib/issues/1039#issuecomment-409312475 will be effective also on Monaparty.
